File: 1.1-Is-Unique/Is-Unique.py
#  Implement an algorithm to determine if a string has all unique characters. What if you cannot use additional data structures?

def isUnique(str):
  newDict = {}
  for x in str:
    if x in newDict.keys():
      return False
    else:
      newDict[x] = 0
  return True

print(isUnique('what'))
print(isUnique('straight'))
print(isUnique('oneMoretest'))
print(isUnique(' '))

File: 1.1-Is-Unique/Is-Unique.rb
# Implement an algorithm to determine if a string has all unique characters. What if you cannot use additional data structures?

def isUnique(str)
  newHash = {}
  str.each_char { |char|
    if newHash.key?(char)
      return false
    else
      newHash[char] = 0
    end
  }
  return true
end

puts isUnique('what')
puts isUnique('straight')
puts isUnique('oneMoretest')
puts isUnique(' ')